generate_assets_metadata.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+
+# Generates the metadata of the assets (ai and wasm). If an asset of the `data_assets` array
+# contains a `chunk_size` key, the script splits the asset into chunks where each chunk has
+# a maximum size of `chunk_size`. The format of the `chunk_size` value is equivalent to the
+# `SIZE` argument in `split` or `gsplit` on macOS. See `split`/`gsplit` man page for more details.
+#
+# The script needs to be executed in the root of the repository.
+#
+# Usage:
+# ./generate_assets_metadata <path of assets_manifest.json> [<wasm version>] [<path of the wasm output directory>]
+
+set -e
+
+OUT_DIR="out"
+ASSETS_METADATA_PATH=$OUT_DIR/assets_metadata.json
+
+if [ -z ${GITHUB_ACTIONS} ]; then
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+        if [ -x "$(command -v gsplit)" ]; then
+            SPLIT="gsplit"
+        else
+            echo "Requires the GNU version of 'split'. Use 'brew install coreutils' to install it."
+            exit 1
+        fi
+    else
+        SPLIT="split"
+    fi
+else
+    if [ $RUNNER_OS == "macOS" ]; then
+        if [ -x "$(command -v gsplit)" ]; then
+            SPLIT="gsplit"
+        else
+            echo "Cannot find 'gsplit'."
+            exit 1
+        fi
+    else
+        SPLIT="split"
+    fi
+fi
```

Comment:
Can we remove the outer if? 
```suggestion
if [[ "$OSTYPE" == "darwin"*  || $RUNNER_OS == "macOS" ]]; then
        if [ -x "$(command -v gsplit)" ]; then
            SPLIT="gsplit"
        else
            echo "Requires the GNU version of 'split'. Use 'brew install coreutils' to install it."
            exit 1
        fi
else
        SPLIT="split"
fi
```